modules/cnf-programming-the-n3000-operator.adoc

```diff
@@ -0,0 +1,349 @@
++
+[NOTE]
+====
+The N3000 daemon updates the FPGA user bitstream using the Open Programmable Acceleration Engine (OPAE) tools and resets the PCI device.
+The update of the FPGA user bitstream may take up to 40 minutes per card. For programming cards on multiple nodes, the programming happens one node at a time.
+====
+.. Apply the CR.
++
+[source,terminal]
+----
+$ oc apply -f n3000-cluster.yaml
+----
++
+The N3000 daemon starts programming the bitstream after the appropriate 5G FEC user bitstream has been provisioned, such as `20ww27.5-2x2x25G-5GLDPC-v1.6.1-3.0.0_unsigned.bin` in this example, and after the CR has been created.
+
+.. Check the status:
++
+[source,terminal]
+----
+oc get n3000node
+----
++
+.Example output
+[source,terminal]
+----
+NAME             FLASH
+node1            InProgress
+----
+
+. Check the logs:
+.. Determine the pod name of the N3000 daemon:
++
+[source,terminal]
+----
+$ oc get pod -o wide | grep n3000-daemonset | grep node1
+----
++
+.Example output
+
+[source,terminal]
+----
+n3000-daemonset-5k55l              1/1     Running   0          15d
+----
+.. View the logs:
++
+[source,terminal]
+----
+$ oc logs n3000-daemonset-5k55l
+----
++
+.Example output
+
+[source,terminal]
+----
+...
+{"level":"info","ts":1608054338.8866854,"logger":"daemon.drainhelper.cordonAndDrain()","msg":"node drained"}
+{"level":"info","ts":1608054338.8867319,"logger":"daemon.drainhelper.Run()","msg":"worker function - start"}
+{"level":"info","ts":1608054338.9003832,"logger":"daemon.fpgaManager.ProgramFPGAs","msg":"Start program","PCIAddr":"0000:1b:00.0"}
+{"level":"info","ts":1608054338.9004142,"logger":"daemon.fpgaManager.ProgramFPGA","msg":"Starting","pci":"0000:1b:00.0"}
+{"level":"info","ts":1608056309.9367146,"logger":"daemon.fpgaManager.ProgramFPGA","msg":"Program FPGA completed, start new power cycle N3000 ...","pci":"0000:1b:00.0"}
+{"level":"info","ts":1608056333.3528838,"logger":"daemon.drainhelper.Run()","msg":"worker function - end","performUncordon":true}
+...
+----
++
+The log file indicates the following flow of events:
+
+* The bitstream is downloaded and validated.
+* The node is drained and no workload is able to run during this time.
+* Flashing is started:
+** The bitstream is flashed into the card.
+** The bitstream is applied.
+* After flashing is complete the PCI device or devices on the node or nodes are reloaded. The Intel wireless Operator is now able to find the new flashed device or devices.
+
+. Verify the status after the FPGA user bitstream update is complete:
++
+[source,terminal]
+----
+oc get n3000node
+----
++
+.Example output
+[source,terminal]
+----
+NAME             FLASH
+node1            Succeeded
+----
+
+. Verify that the bitstream ID of the card has changed:
++
+[source,terminal]
+----
+oc get n3000node node1 -o yaml
+----
++
+.Example output
+[source,yaml]
+----
+status:
+      conditions:
+          - lastTransitionTime: "2020-12-15T18:18:53Z"
+          message: Flashed successfully <1>
```

Comment:
indetation issue here everything needs to be on the same line as `lastTransitionTime`